assessments/projects/linkerd/self-assessment.md

```diff
@@ -0,0 +1,321 @@
+together, this document and the joint-assessment serve as a cornerstone for if and when
+Linkerd seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+### Critical
+
+**Data Plane**
+
+The data plane is responsible for gathering metrics, observing traffic, and applying policy. It is comprised of sidecar proxies which intercept application network calls and implement rules and logic. Linkerd uses a service mesh specific Linkerd2-proxy, an ultra-light and transparent micro-proxy.
+
+**Control Plane**
+
+The control plane of a service mesh provides the command and control signals required for the data plane to operate. The control plane controls the data plane and provides the UI and API that operators use to configure, monitor, and operate the mesh. The control plane includes three components, destination and identity services and a proxy injector. The destination service fetches policy, discovery, and profile information on requests. The identity service implements mTLS (mutually-authenticated Transport Layer Security) by accepting CSRs and returning signed certificates. Lastly, the proxy injector is a Kubernetes admission controller mutator that adds containers to annotated pods. 
+
+### Security Relevant
+**CLI**
+
+The CLI (optional) can be used to interact with the data and control planes. It is run on the local machine, outside of the Kubernetes cluster to interact with Linkerd. 
+
+**Viz Extension**
+
+To access on-stack metrics and CLI tools and dashboards, the viz extension must be installed. This extension installs the following components into the linkerd-viz namespace:
+
+* a Prometheus instance
+* metrics-api, tap, tap-injector, and web components
+
+Metrics are stored transiently to limit resources.
+The tap tool supports real time analysis of live traffic. This feature can potentially expose sensitive data such as request and response headers. The tap resource used Role-based access control (RBAC) authorization. This method regulates access to individual users within an organization based on their specified role. 
+
+* RBAC API prevents privilege escalation by editing roles or role bindings
+* wildcards in resource and verb entries could result in overly permissive access being granted to sensitive resources
+* the principle of least privilege, in which minimal rights are applied to users and accounts, should be applied
+* granting cluster-admin permissions to the "default" service account in the kube-system namespace contains Secrets that grant super-user access to your cluster's API
+* granting super-user access to all service accounts cluster-wide allows any application full access to your cluster, and also grants any user with read access to Secrets (or the ability to create any pod) full access to your cluster
+
+
+## Project compliance
+
+## Security and Vulnerability Management:
+
+### Vulnerability Reporting: 
```

Comment:
I do not follow what vulnerability reporting in this context is, and how sidecar/microservice architecture is aiding to it. It doesn't seem to me like there is coherence between the concepts mentioned in this section. 
Please clarify.

Reply:
Thank you, I've updated that section to report on how Linkerd reports vulnerabilities from a project standpoint.